src/Microsoft.Sbom.Adapters/Adapters/ComponentDetection/ScannedComponentExtensions.cs

```diff
@@ -34,6 +35,7 @@ public static class ScannedComponentExtensions
             PipComponent pipComponent => pipComponent.ToSbomPackage(component?.License),
             PodComponent podComponent => podComponent.ToSbomPackage(component?.License),
             RubyGemsComponent rubyGemsComponent => rubyGemsComponent.ToSbomPackage(component?.License),
+            SpdxPackageComponent spdxPackageComponent => spdxPackageComponent.ToSbomPackage(component?.License),
```

Comment:
We intentionally leave out the spdx components because we display them in the "External References" section of the SBOM. So this can be removed.

Reply:
I am not sure, but when I tested, it just display Spdx file info in the External References and not the packages inside the packages itself. Am I missing anything here or it needs to be enabled?

Reply:
Ah, I see. It looks like you are correct but regardless this is a duplicate of https://github.com/microsoft/sbom-tool/blob/main/src/Microsoft.Sbom.Adapters/Adapters/ComponentDetection/SpdxComponentExtensions.cs

Reply:
okay.. My folder structure looks like

C drive -> 
   1) ProjectA
       a) src 
       b) ProjectBSbom
           A) manifest.spdx
           
   2) ProjectB 
        a) src
        b) sbom
           A) manifest.spdx
           
 Project B sbom generated using cmd
 
**sbom-tool-win-x64 generate -b "C:\ProjectB" -bc "C:\ProjectB" -pn "ProjectB" -pv "1.0.1" -ps "XYZ" -nsb "https://XYX.com" -d true**


Now, I want to make use of Project B Sbom in Project A, So, i copies Project B-> SBOM in Project A Folder
 
**sbom-tool-win-x64 generate -b "C:\ProjectA" -bc "C:\ProjectA" -pn "ProjectA" -pv "1.0.1" -ps "XYZ" -nsb "https://XYX.com" -d true**

and it does find the file in externalDocumentRefs but there is no package information from Project B-> SBOM to Project A SBOM.

Reply:
I tested this as well. It seems you are correct in that it does not show up in the packages section due to other reasons but regardless there is no need for this additional function. SPDX components are already filtered out and handled by the time execution reaches this point.

Reply:
i did not get it completely.. do you mean that SPDX will be fixed in the future or there is a bug which restrict packages to not list in the spdx file. Could you please clarify?

Reply:
No, this behavior is by design. We do not want spdx components in the packages section of the SBOM. We already display the relevant information in the externalDocumentRefs.

Reply:
okay.. Thanks for letting me know that it is by design. however i just wanted to understand one more thing here is that

consider below reference SPDX file generated for ProjectA which include spdx document from ProjectA as well. (Please ignore dummy values)

{
  "files": [
    {
      "fileName": "./SBOM/ProjectB/SBOM.spdx.json",
      "SPDXID": "SPDXRef-File--SBOM-ProjectB-SBOM.spdx.json-abcdeqwertyuiopasdfghjklzxcvbnm123456789",
      "checksums": [
        {
          "algorithm": "SHA256",
          "checksumValue": "abcdeqwertyuiopasdfghjklzxcvbnm123456789"
        }
      ],
      "licenseConcluded": "NOASSERTION",
      "licenseInfoInFiles": [
        "NOASSERTION"
      ],
      "copyrightText": "NOASSERTION",
      "fileTypes": [
        "SPDX"
      ]
    }
  ],
  "packages": [
    {
      "name": "SQLite.SQLite3.Native",
      "SPDXID": "SPDXRef-Package-abcdeqwertyuiopasdfghjklzxcvbnm123456789",
      "downloadLocation": "NOASSERTION",
      "filesAnalyzed": false,
      "licenseConcluded": "NOASSERTION",
      "licenseDeclared": "NOASSERTION",
      "copyrightText": "NOASSERTION",
      "versionInfo": "1.0.1",
      "externalRefs": [
        {
          "referenceCategory": "PACKAGE-MANAGER",
          "referenceType": "purl",
          "referenceLocator": "pkg:nuget/SQLite.SQLite3.Native@1.0.1"
        }
      ],
      "supplier": "Organization: SQLite Org"
    }
  ],
  "externalDocumentRefs": [
    {
      "externalDocumentId": "DocumentRef-ProjectB-1.0.1-abcdeqwertyuiopasdfghjklzxcvbnm123456789",
      "spdxDocument": "https://XYZCoorporation.com/ProjectB/1.0.1/ablWjLw0nkmg2IZlJuXpwA",
      "checksum": {
        "algorithm": "SHA1",
        "checksumValue": "abcdeqwertyuiopasdfghjklzxcvbnm123456789"
      }
    }
  ],
  "relationships": [
    {
      "relationshipType": "PREREQUISITE_FOR",
      "relatedSpdxElement": "DocumentRef-ProjectB-1.0.1-abcdeqwertyuiopasdfghjklzxcvbnm123456789:SPDXRef-RootPackage",
      "spdxElementId": "SPDXRef-RootPackage"
    }
  ],
  "spdxVersion": "SPDX-2.2",
  "dataLicense": "xyz-1.0",
  "SPDXID": "SPDXRef-DOCUMENT",
  "name": "ProjectA 1.0.1",
  "documentNamespace": "https://XYZCoorporation.com/ProjectA/1.0.1/asdfghjklertyu--12345678",
  "creationInfo": {
    "created": "2023-09-19T05:58:13Z",
    "creators": [
      "Organization: XYZCoorporation",
      "Tool: Microsoft.SBOMTool-1.5.2-preview.0.45+abcdeqwertyuiopasdfghjklzxcvbnm123456789"
    ]
  },
  "documentDescribes": [
    "SPDXRef-RootPackage"
  ]
}


my question is that where should we display the packages information from Spdx file  of Project A?
1) Should it be display in externalDocumentRefs section?
2) Or should it be display in package section. (I also see there is externalRefs attribute in package section but i believe this is to display PURL or any other type info for that package)?

Reply:
We don't want to display information from within the SPDX file. SBOMs are representative of an individual piece of software. We just list the presence of an SPDX file in the externalDocumentRefs section. No additional details are required so this, along with the other spdx related changes in this PR should be removed. Thank you, for your contributions.

Reply:
@sebasgomez238 i have reverted changes related to spdx package component. Thanks for clarification :)